query/query_runner.pl

```diff
@@ -0,0 +1,40 @@
+
+main(Query,Filename):-
+	consult(Filename),
+	run_query(Query).
+
+run_query(Query):-
+	findall(Result, (atom_to_term(Query, Term, Result), Term), ResultSet),
+	prettyWriteResultSet(ResultSet).
+
+prettyWriteResultSet([]):-
+	writeln('false.').
+
+prettyWriteResultSet([OneResult]):-
+	prettyWriteOneResult(OneResult),
+	writeln('.').
+
+prettyWriteResultSet([OneResult | ResultSet]):-
+	prettyWriteOneResult(OneResult),
+	writeln(' ;'),
+	prettyWriteResultSet(ResultSet).
+
+
+prettyWriteOneResult([]):-
+	write('true').
+
+prettyWriteOneResult([OneBinding]):-
+	writeBinding(OneBinding).
+
+prettyWriteOneResult([OneBinding | OneResult]):-
+	writeBinding(OneBinding),
+	writeln(','),
+	prettyWriteOneResult(OneResult).
+
+writeBinding(OneBinding):-
+	compound_name_arguments(OneBinding, (=), [VarName, Value]),
+	writef('%w = %w', [VarName, Value]).
+
+writeBinding(NotABinding):-
+	writef('ERROR: writeBinding/1: Expected Binding, but no equals was founded in: %w\n', [NotABinding]),
+	halt.
```

Comment:
Actually it won't, because a goal executed with prolog this way behaves differently from the interactive mode, when it finds a path that applies it doesn't go any further, so it will only enter the second clause if the argument doesn't fulfil the first one. 
But thinking about it, making it mutually exclusive isn't that hard and maybe it will make it more expressive.